backend/app/api/deps.py

```
@@ -10,21 +10,32 @@
+        query = select(Users).where(
+            Users.user_id == int(user_id),
+            Users.is_active == True,
+            Users.deleted_at.is_(None)
+        )
+        result = await db.execute(query)
+        user = result.scalar_one_or_none()
+        if not user:
+            raise HTTPException(status_code=401, detail="User not found")
+        return user
+    except Exception:
+        raise HTTPException(status_code=401, detail="Invalid credentials")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)